app/src/pages/inside/testCaseLibraryPage/testCaseListWrapper/testCaseListWrapper.tsx

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2025 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState, useCallback, useMemo } from 'react';
+import { useIntl } from 'react-intl';
+import classNames from 'classnames/bind';
+import { Pagination } from '@reportportal/ui-kit';
+import { TestCaseList } from 'components/testCaseList';
+import { TestCase } from 'components/testCaseList/testCaseCard/testCaseCard';
+import { ITEMS_PER_PAGE_OPTIONS } from 'components/testCaseList/mockData';
+import {
+  DEFAULT_CURRENT_PAGE,
+  DEFAULT_ITEMS_PER_PAGE,
+} from 'components/testCaseList/testCaseList.constants';
+import { messages } from '../messages';
+import styles from './testCaseListWrapper.scss';
+
+const cx = classNames.bind(styles);
+
+interface TestCaseListWrapperProps {
+  testCases: TestCase[];
+  loading: boolean;
+  setTestCases: (testCases: TestCase[] | ((prev: TestCase[]) => TestCase[])) => void;
+}
+
+export const TestCaseListWrapper = ({
+  testCases,
+  loading,
+  setTestCases,
+}: TestCaseListWrapperProps) => {
+  const { formatMessage } = useIntl();
+  const [activePage, setActivePage] = useState<number>(DEFAULT_CURRENT_PAGE);
+  const [pageSize, setPageSize] = useState<number>(DEFAULT_ITEMS_PER_PAGE);
+  const [filteredTestCases, setFilteredTestCases] = useState<TestCase[]>([]);
+  const [searchValue, setSearchValue] = useState<string>('');
+
+  const updateFilteredTestCases = useCallback(
+    (searchTerm: string = searchValue) => {
+      let filtered = testCases;
+
+      if (searchTerm.trim()) {
+        const term = searchTerm.toLowerCase();
+        filtered = filtered.filter(
+          (testCase) =>
+            testCase.name.toLowerCase().includes(term) ||
+            testCase.tags.some((tag) => tag.toLowerCase().includes(term)),
+        );
+      }
+
+      setFilteredTestCases(filtered);
+      setActivePage(DEFAULT_CURRENT_PAGE);
+    },
+    [testCases, searchValue],
+  );
+
+  useMemo(() => {
```

Comment:
I am really concerned here:
* it runs update function
* memoizes nothing

remove it please